esp-hal/src/i2c/master/mod.rs

```diff
@@ -1578,7 +1597,7 @@ impl Driver<'_> {
         };
         let scl_high = half_cycle - scl_wait_high;
         let sda_hold = half_cycle / 4;
-        let sda_sample = half_cycle / 2 + scl_wait_high;
+        let sda_sample = half_cycle / 2;
```

Comment:
To be fair, I should probably remove the `half_cycle/2` term instead. Technically, we can sample at any point after the rising edge, and `scl_wait_high` is supposed to be somewhat dependent on the hardware (i.e. it should be tunable, maybe). `scl_wait_high` represents the rising edge of the clock signal, after which the clock signal is high for `scl_high`. We should probably aim at the start/middle of the interval covered by `scl_high`.